kodilla-good-patterns/src/main/java/com/kodilla/good/patterns/flight_finder/PrivateJourney.java

```diff
@@ -1,26 +1,66 @@
 package com.kodilla.good.patterns.flight_finder;
 
+import java.util.*;
+
 public final class PrivateJourney implements Journey
 {
-    private final String initialCity;
-    private final String destinationCity;
+    private final LinkedList<Flight> journeyPlan;
 
-    public PrivateJourney(final String initialCity, final String destinationCity)
+    public PrivateJourney(final LinkedList<Flight> journeyPlan)
+    {
+        this.journeyPlan = journeyPlan;
+    }
+
+    @Override
+    public String getInitialCity() throws NoSuchElementException
     {
-        this.initialCity = initialCity;
-        this.destinationCity = destinationCity;
+        try
+        {
+            return this.journeyPlan.getFirst().getDepartureCity();
+        }
+        catch (NoSuchElementException e)
+        {
+            return "";
+        }
+    }
+
+    @Override
+    public String getDestinationCity() throws NoSuchElementException
+    {
+        try
+        {
+            return this.journeyPlan.getLast().getArrivalCity();
+        }
+        catch (NoSuchElementException e)
+        {
+            return "";
+        }
+
     }
 
     @Override
-    public String getInitialCity()
+    public boolean isLooped()
     {
-        return this.initialCity;
+        Set<String> listOfCities = new HashSet<>();
+        for(Flight flight:this.journeyPlan)
+        {
+            listOfCities.add(flight.getDepartureCity());
+            listOfCities.add(flight.getArrivalCity());
+        }
+        if(listOfCities.size()==this.journeyPlan.size()+1)
```

Comment:
można napisać po prostu:
return listOfCities.size() == this.journeyPlan.size() + 1;

Reply:
Wolę zostawić if-a dla zwrócenia uwagi, że kod po if-e będzie wykonany i nie musi być zawarty w bloku else.